app/models/participant_declaration.rb

```diff
@@ -45,6 +46,8 @@ class ParticipantDeclaration < ApplicationRecord
   delegate :fundable?, to: :participant_profile, allow_nil: true
 
   validates :course_identifier, :user, :cpd_lead_provider, :declaration_date, :declaration_type, :cohort, presence: true
+  validates :voided_by_user, presence: true, if: :voided_at
+  validates :voided_at, presence: true, if: :voided_by_user
```

Comment:
Doesn't it create a dead lock condition? Not sure.. as it validates the other when the first present and vice-versa?

Reply:
Not sure I follow, I can't see how it could deadlock? It'll process the validation rules in order